nodeadm/internal/kubelet/config.go

```diff
@@ -437,3 +455,42 @@ func getNodeIp(ctx context.Context, imdsClient *imds.Client, cfg *api.NodeConfig
 		return "", fmt.Errorf("invalid ip-family. %s is not one of %v", ipFamily, []api.IPFamily{api.IPFamilyIPv4, api.IPFamilyIPv6})
 	}
 }
+
+func getCPUMillicoresToReserve() int {
+	totalCPUMillicores, err := util.GetMilliNumCores()
+	if err != nil {
+		zap.L().Error(fmt.Sprintf("Error found when GetMilliNumCores: %v", err))
+		return 0
+	}
+	cpuRanges := []int{0, 1000, 2000, 4000, totalCPUMillicores}
+	cpuPercentageReservedForRanges := []int{600, 100, 50, 25}
+	cpuToReserve := 0
+
+	for i, percentageToReserveForRange := range cpuPercentageReservedForRanges {
+		startRange := cpuRanges[i]
+		endRange := cpuRanges[i+1]
+		cpuToReserve += getResourceToReserveInRange(totalCPUMillicores, startRange, endRange, percentageToReserveForRange)
+	}
+
+	return cpuToReserve
+}
+
+// getResourceToReserveInRange calculates the CPU resources to reserve for a given range.
+func getResourceToReserveInRange(totalCPU, startRange, endRange, percentage int) int {
+	if totalCPU > startRange {
+		if totalCPU < endRange {
+			return (totalCPU - startRange) * percentage / 10000
+		}
+		return (endRange - startRange) * percentage / 10000
+	}
+	return 0
```

Comment:
super nit, but this isn't instantly readable to me, could it be 
```suggestion
	if totalCPU <= startRange {
	    return 0
	}
	reserved := totalCPU
	if reserved > endRange {
	    reserved = endRange
	}
	return (reserved - startRange) * percentage / 10000
```

Reply:
ack, updated